scene/resources/skeleton_modification_2d.cpp

```diff
@@ -0,0 +1,3433 @@
+	} else if (path.begins_with("constraint_angle_max")) {
+		set_constraint_angle_max(Math::deg2rad(float(p_value)));
+	} else if (path.begins_with("constraint_angle_invert")) {
+		set_constraint_angle_invert(p_value);
+	} else if (path.begins_with("constraint_in_localspace")) {
+		set_constraint_in_localspace(p_value);
+	} else if (path.begins_with("additional_rotation")) {
+		set_additional_rotation(Math::deg2rad(float(p_value)));
+	}
+
+#ifdef TOOLS_ENABLED
+	if (path.begins_with("editor/draw_gizmo")) {
+		set_editor_draw_gizmo(p_value);
+	}
+#endif // TOOLS_ENABLED
+
+	return true;
+}
+
+bool SkeletonModification2DLookAt::_get(const StringName &p_path, Variant &r_ret) const {
+	String path = p_path;
+
+	if (path.begins_with("enable_constraint")) {
+		r_ret = get_enable_constraint();
+	} else if (path.begins_with("constraint_angle_min")) {
+		r_ret = Math::rad2deg(get_constraint_angle_min());
+	} else if (path.begins_with("constraint_angle_max")) {
+		r_ret = Math::rad2deg(get_constraint_angle_max());
+	} else if (path.begins_with("constraint_angle_invert")) {
+		r_ret = get_constraint_angle_invert();
+	} else if (path.begins_with("constraint_in_localspace")) {
+		r_ret = get_constraint_in_localspace();
+	} else if (path.begins_with("additional_rotation")) {
+		r_ret = Math::rad2deg(get_additional_rotation());
+	}
+
+#ifdef TOOLS_ENABLED
+	if (path.begins_with("editor/draw_gizmo")) {
+		r_ret = get_editor_draw_gizmo();
+	}
+#endif // TOOLS_ENABLED
+
+	return true;
+}
+
+void SkeletonModification2DLookAt::_get_property_list(List<PropertyInfo> *p_list) const {
+	p_list->push_back(PropertyInfo(Variant::BOOL, "enable_constraint", PROPERTY_HINT_NONE, "", PROPERTY_USAGE_DEFAULT));
+	if (enable_constraint) {
+		p_list->push_back(PropertyInfo(Variant::FLOAT, "constraint_angle_min", PROPERTY_HINT_RANGE, "-360, 360, 0.01", PROPERTY_USAGE_DEFAULT));
+		p_list->push_back(PropertyInfo(Variant::FLOAT, "constraint_angle_max", PROPERTY_HINT_RANGE, "-360, 360, 0.01", PROPERTY_USAGE_DEFAULT));
+		p_list->push_back(PropertyInfo(Variant::BOOL, "constraint_angle_invert", PROPERTY_HINT_NONE, "", PROPERTY_USAGE_DEFAULT));
+		p_list->push_back(PropertyInfo(Variant::BOOL, "constraint_in_localspace", PROPERTY_HINT_NONE, "", PROPERTY_USAGE_DEFAULT));
+	}
+	p_list->push_back(PropertyInfo(Variant::FLOAT, "additional_rotation", PROPERTY_HINT_NONE, "", PROPERTY_USAGE_DEFAULT));
+
+#ifdef TOOLS_ENABLED
+	if (Engine::get_singleton()->is_editor_hint()) {
+		p_list->push_back(PropertyInfo(Variant::BOOL, "editor/draw_gizmo", PROPERTY_HINT_NONE, "", PROPERTY_USAGE_DEFAULT));
+	}
+#endif // TOOLS_ENABLED
+}
+
+void SkeletonModification2DLookAt::execute(float delta) {
+	ERR_FAIL_COND_MSG(!stack || !is_setup || stack->skeleton == nullptr,
+			"Modification is not setup and therefore cannot execute!");
+	if (!enabled) {
+		return;
+	}
+
+	if (target_node_cache.is_null()) {
+		_print_execution_error(true, "Target cache is out of date. Attempting to update...");
```

Comment:
Why don't relay on `ERR_FAIL_COND` and `ERR_PRINT`?
Using custom methods inside the nodes, could make future code refactoring more difficult. Much better stick to Godot standard.